Dispatcher/UI/Screens/HomePage/HomePageViewController.swift

```diff
@@ -0,0 +1,58 @@
+import UIKit
+
+class HomePageViewController: UIViewController {
+    
+    @IBOutlet weak var tableView: UITableView!
+    var viewModel: HomePageViewModel = HomePageViewModel(repository: MockArticleRepository())
+    var mockArticles:[Card] = []
+    let navigationBar = NavigationBar()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        mockArticles = viewModel.repository.getArticles()
+        initNavigationBar()
+        initTableView()
+    }
+    
+    func initNavigationBar() {
+        navigationBar.delegate = self
+        navigationBar.setupNavigationBar(for: self)
+    }
+    
+    func initTableView() {
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.register(UINib(nibName:String(describing: ArticleCell.self), bundle: nil), forCellReuseIdentifier: ArticleCell.identifier)
+        tableView.rowHeight = 449
+    }
+}
+
+extension HomePageViewController: UITableViewDataSource, UITableViewDelegate {
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return mockArticles.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: ArticleCell.identifier, for: indexPath)
+        as? ArticleCell
+        let item = mockArticles[indexPath.row]
```

Comment:
what if you were to replace the mockRepostiroy for a regular repositroy
this will be misleading because it wont be mock articles anymore
this is creating tight coupling